src/main/java/decodes/cwms/CwmsConnectionPool.java

```diff
@@ -0,0 +1,399 @@
+        info.setConnection(conn);
+        info.setDbOfficeCode(officeId2code(conn, officeId));
+
+        // MJM 2018-2/21 Force autoCommit on.
+        try{ conn.setAutoCommit(true); }
+        catch(SQLException ex)
+        {
+            log.warning("Cannot set SQL AutoCommit to true: " + ex);
+        }
+
+        String priv = getOfficePrivileges(conn, info.getLoginInfo().getUrl(), officeId);
+        setCtxDbOfficeId(conn, officeId, info.getDbOfficeCode(), priv);
+
+
+        try
+		{
+            String name = String.format("CwmsConnectionPool(%s/%s)",info.getLoginInfo().getUrl(),info.getLoginInfo().getUser());
+			ManagementFactory.getPlatformMBeanServer()
+							 .registerMBean(this, new ObjectName("org.opendcs:type=TimeSeriesDb,name=\""+name+"\""));
+		}
+		catch(JMException ex)
+		{
+			log.warning("Unable to register tracking bean " + ex.getLocalizedMessage());
+		}
+
+    }    
+
+    @Override
+	public int getConnectionsOut()
+    {
+		return connectionsOut.size();
+	}
+
+	@Override
+	public int getConnectionsAvailable()
+    {
+		return 10 - connectionsOut.size();
+	}
+
+	@Override
+	public String getThreadName()
+    {
+		return Thread.currentThread().getName();
+	}
+
+	@Override
+	public int getGetConnCalled()
+    {
+		return this.connectionsRequested;
+	}
+
+	@Override
+	public int getFreeConnCalled()
+    {
+		return this.connectionsFreed;
+	}
+
+	@Override
+	public int getUnknownReturned()
+    {
+		return this.unknownConnReturned;
+	}
+
+	@Override
+	public TabularData getConnectionsList() throws OpenDataException
+    {
+		TabularData td = new TabularDataSupport(JMXTypes.CONNECTION_LIST);
+        for(WrappedConnection conn: this.connectionsOut)
+        {
+            td.put(conn.asCompositeData());
+        }
+        return td;
+	}
+
+
+    public Connection getConnection() throws SQLException
+    {
+        connectionsRequested++;
+        Connection conn = pool.getConnection(info.getLoginInfo());
+        WrappedConnection wc = new WrappedConnection(conn,(c)->CwmsDbConnectionPool.close(c),Optional.of(this));
+        connectionsOut.add(wc);        
+        return wc;
+    }
+
+    public void returnConnection(Connection conn) throws SQLException
+    {
+        if(connectionsOut.contains(conn))
+        {
+            connectionsFreed++;
+            connectionsOut.remove(conn);
+            conn.close();
+
+        }
+        else
+        {
+            unknownConnReturned++;
+        }     
+    }
+
+
+
+    private String getOfficePrivileges(Connection conn, String user, String dbOfficeId) throws BadConnectException
+	{
+		String ret = null;
+		try
+		{
+			ArrayList<StringPair>officePrivileges = determinePrivilegedOfficeIds(conn);
+			// MJM 2018-12-05 now determine the highest privilege level that this user has in
+			// the specified office ID:
+			log.finest("Office Privileges for user '" + user + "'");
+			for(StringPair op : officePrivileges)
+			{
+				if (op == null)
```

Comment:
logging at warning seems pretty high for these statements, especially without any context in the message.